src/aws/osml/model_runner/database/ddb_helper.py

```diff
@@ -87,22 +91,87 @@ def get_ddb_item(self, ddb_item: DDBItem) -> Dict[str, Any]:
         """
         return self.table.get_item(Key=self.get_keys(ddb_item=ddb_item))["Item"]
 
-    def put_ddb_item(self, ddb_item: DDBItem, condition_expression: str = None) -> Dict[str, Any]:
+    def put_ddb_item(
+        self, ddb_item: DDBItem, condition_expression: str = None, max_retries=30, initial_delay=0.5, max_delay=5
+    ) -> Dict[str, Any]:
         """
         Put a DynamoDB item into the table
 
         :param ddb_item: DDBItem = item that we want to put (required)
         :param condition_expression: str = condition that must be satisfied in order for a
                                             conditional PutItem operation to succeed.
+        :param max_retries: int = Maximum number of retries for unprocessed items.
+        :param initial_delay: float = Initial delay in seconds for the first retry.
+        :param max_delay: float = Maximum delay in seconds between retries.
 
         :return: Dict[str, Any] = item from the put_item response
         """
-        if condition_expression:
-            response = self.table.put_item(Item=ddb_item.to_put(), ConditionExpression=condition_expression)
-        else:
-            response = self.table.put_item(Item=ddb_item.to_put())
+        retries = 0
+        delay = initial_delay
+
+        while retries < max_retries:
```

Comment:
Why are you implementing custom retry logic here? The AWS DDB SDK already provides implementation of retries with an exponential backoff. See: https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Programming.Errors.html#Programming.Errors.RetryAndBackoff

Reply:
You'll notice this isn't actually an exponential back-off ( which was causing an over exaggerated lag for some writes )  but a jitter based retry method which from testing seemed to work much better.

Reply:
And too your point I removed the custom retry logic outside of retrying unprocessed items.